lib/metadata-storage.ts

```diff
@@ -0,0 +1,121 @@
+import { u8aToString } from "@polkadot/util";
+import { u8aConcat } from "@polkadot/util/u8a";
+import { MetadataStorage, createStorage } from "@zeitgeistpm/sdk";
+import { Codec, JsonCodec } from "@zeitgeistpm/utility/dist/codec";
+import * as O from "@zeitgeistpm/utility/dist/option";
+import * as Te from "@zeitgeistpm/utility/dist/taskeither";
+import { Storage, StorageError } from "@zeitgeistpm/web3.storage";
+import * as IPFSHTTPClient from "ipfs-http-client";
+import { CID } from "multiformats/cid";
+
+const node = IPFSHTTPClient.create({ url: "https://ipfs.zeitgeist.pm" });
+
+export const createMetadataStorage = (): MetadataStorage => {
```

Comment:
When is this expected to be used and when is the cluster expected to be used?

Reply:
This is only used to read data from IPFS so this should be set to the new hosted ipfs node I believe. Cluster is used to add/pin files and remove them when rolling back a failed transaction.

Reply:
We will shortly shut down the cluster and only use a singular hosted IPFS node (replication happens also in Subsquid). For now this is the right approach though, let's adjust once we have the hosted IPFS node ready.